### PR TITLE
Add duplicate ASGs for public networks and dns to apply individually

### PR DESCRIPTION
**Pull request by apburnes**
### Supports the work in #565

- [x] Add duplicate ASGs for dns and public_networks to be applied to each space individually

## security considerations
None

### CODE REVIEW

**Review by bengerman13 — APPROVED** [review]

**Comment by apburnes** [comment]
Closing to work on transition steps in #568

**Comment by apburnes** [comment]
Ignore previous. Wrong issue.

**Comment by apburnes** [comment]
Remove `- [ ] Add acceptance tests to test app egress rules` from this PR and create new issue to better outline formal test suite.

File: terraform/asg.tf [hunk 1]
```diff
@@ -76,6 +107,24 @@ resource "cloudfoundry_asg" "dns" {
   }
 }
 
+# New dns asg to apply to spaces individually, not globally.
+
+resource "cloudfoundry_asg" "dns_egress" {
```
Comment (apburnes) [inline terraform/asg.tf:112]: Do we need to apply a non global DNS asg? [ADR](https://github.com/cloud-gov/product/pull/1605/files) does not mention it. Would there be a use case to not allow DNS to resolve for a space?

**Review by bengerman13 — APPROVED** [review]

**Pull request merged**